File: mobile/lib/shared/models/app_user.dart
class AppUser {
  const AppUser({
    required this.id,
    required this.name,
    required this.email,
    required this.currency,
    this.avatar,
  });

  final int id;
  final String name;
  final String email;
  final String currency;
  final String? avatar;

  factory AppUser.fromJson(Map<String, dynamic> json) {
    return AppUser(
      id: json['id'] as int,
      name: json['name'] as String,
      email: json['email'] as String,
      currency: json['currency'] as String,
      avatar: json['avatar'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {'id': id, 'name': name, 'email': email, 'currency': currency, 'avatar': avatar};
  }
}

File: appReact/src/shared/models/appUser.ts
export interface AppUser {
  id: number;
  name: string;
  email: string;
  currency: string;
  avatar: string | null;
}

export function appUserFromJson(json: any): AppUser {
  return {
    id: json.id,
    name: json.name,
    email: json.email,
    currency: json.currency,
    avatar: json.avatar ?? null,
  };
}
